### PR TITLE
DCNG-784 Provision a GCP Kubernetes cluster

### PR DESCRIPTION
**Pull request by pbruski**

### CODE REVIEW

**Review by kennymacleod — APPROVED** [review]

File: src/test/scripts/helm_install.sh [hunk 1]
```diff
@@ -6,11 +6,17 @@ set -x
 getCurrentClusterType() {
```
Comment (kcichy-atlassian) [inline src/test/scripts/helm_install.sh:6]: TIL Bash is even worse than I used to think. No return values and returning by echoing 🤦 .

File: src/test/scripts/helm_install.sh [hunk 2]
```diff
@@ -6,11 +6,17 @@ set -x
...
   local serverAddress=$(kubectl config view --minify -o json | jq -r '.clusters[0].cluster.server')
 
-  case "${serverAddress}" in
+  local clusterType=$(case "${serverAddress}" in
     *.eks.amazonaws.com*) echo EKS;;
     *.azmk8s.io*) echo AKS;;
-    *gke*) echo GKE;;
     *.kitt-inf.net*) echo KITT;;
+  esac)
+  [ "$clusterType" ] && { echo $clusterType ; return ; }
```
Comment (kcichy-atlassian) [inline src/test/scripts/helm_install.sh:14]: How about replacing this trick with immediate returns?
```
  case "${serverAddress}" in
    *.eks.amazonaws.com*) echo EKS; return;;
    *.azmk8s.io*) echo AKS; return;;
    *gke*) echo GKE; return;;
    *.kitt-inf.net*) echo KITT; return;;
  esac

  case "$(kubectl config view --minify -o json | jq -r '.clusters[0].name')" in
    gke*) echo GKE; return;;
    *) echo CUSTOM; return;;
  esac
```
?

**Review by kcichy-atlassian — APPROVED** [review]

File: src/test/config/bitbucket/values-AKS.yaml [hunk 3]
```diff
@@ -20,7 +20,7 @@ volumes:
     customVolume:
       persistentVolumeClaim:
         claimName: ${shared.pvc.name} # Pre-provisioned, and shared by all of our pods
-    subPath: ${helm.release.prefix}-bitbucket # Since all of our pods share the same EFS PV, we use subpath mounts to prevent interference
```
[the remark this answers is not in the document]
Comment (pbruski) [reply src/test/config/bitbucket/values-AKS.yaml:23]: Yes, we do - both are AWS-based.

File: src/test/scripts/helm_install.sh (hunk 1 above)
Comment (pbruski) [reply src/test/scripts/helm_install.sh:6]: well, you can return a status. You can also return via a param outpu variable, I don't particularly like that syntax, though.

**Pull request merged**